src/encrypt/mod.rs

```diff
@@ -0,0 +1,74 @@
+use dusk_bls12_381::Scalar;
+
+use std::io;
+
+/// Encapsulates an encrypted data
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct EncryptedData {
+    a: Scalar,
+    b: Scalar,
```

Comment:
I'm not sure that these 2 should be BLS Scalars. If these values are used in ECC ops, or are the result of them, they should be JubJubScalars.

Unless, these two `Scalar`s are representing a JubJubAffinePoint. In that case, I think it would be better to use `jubjub::AffinePoint` instead of this for consistency reasons.

Reply:
These scalars are the inputs used for Hades permutations. They should be BLS, as the standard poseidon hash input.

These values are not expected to perform any ECC ops here. The ECC is required only for the shared key resolution that will be part of the circuit